packages/server-wallet/src/__test__/with-peers/crash-tolerance/crash-tolerance.test.ts

```diff
@@ -85,25 +51,29 @@ it('Create a directly-funded channel between two wallets, of which one crashes m
   });
 
   //    > PreFund0A
-  const resultB0 = await b.pushMessage(getPayloadFor(participantB.participantId, resultA0.outbox));
+  const resultB0 = await peerWallets.b.pushMessage(
+    getPayloadFor(participantB.participantId, resultA0.outbox)
+  );
 
   expect(getChannelResultFor(channelId, resultB0.channelResults)).toMatchObject({
     status: 'proposed',
     turnNum: 0,
   });
 
   // Destory Wallet b and restart
-  b = await crashAndRestart(b);
+  peerWallets.b = await crashAndRestart(peerWallets.b);
```

Comment:
This line is a bit strange: Since `peerWallets` is imported from another module, you are overwriting the value of an import. I think it's _ok_, given my understanding of how the jest test runner works, but it definitely smells a bit off.

Reply:
An alternative is
1. `path/to/with-peers-setup-teardown` exports a `peerWalletsFactory`
2. test files assign `let peerWallets = peerWalletsFactory()` at the top of their file

Reply:
Agreed. My current understanding: Jest test framework uses (abuses?) global state as a tool to supply the minimal necessary context for unit tests to run. Global state is "generally bad", and abstracting global state for these tests to one file tends toward doing something that is "generally bad" but maybe an acceptable trade-off in context of Jest.

I'll give ~25 minutes to providing a cleaner export here (more if requested), but I know I had some issue that discouraged one prior approach.

caveat: scoping and hoisting in JS is sometimes surprising; I in no way claim complete knowledge of the assumptions Jest makes about available scope

Reply:
I know that if we do not set the `runInBand` option, each test suite (file) runs in a separate node process, meaning it's "ok" to modify objects in the global scope.

I also know that we have to run tests with the `runInBand` option, but don't know the implications of that here.